app/controllers/api/admin/commodities_controller.rb

```diff
@@ -25,6 +37,18 @@ def commodity_code
       def productline_suffix
         params[:id].split('-', 2)[1] || '80'
       end
+
+      def all_commodities
+        commodity_groups = []
+
+        # Splitting the queries this way make the execution faster
+        (0..9).each do |starting_digit|
+          commodity_groups << Sequel::Model.db.fetch('select * from public.goods_nomenclature_export_new(?, ?) order by 2, 3', "#{starting_digit}%", actual_date)
```

Comment:
Personally I think this is clearly a complex query so should be encapsulated on a model object, either an existing one (ie `Commodity`), or a separate object if its felt its sufficient unrelated to the Commodity

We should be able to add specs against the query itself then to check it is behaving as expected